### DISCUSSION TITLE
Deconstruct using def

### DISCUSSION THREAD

**Discussion by Chemaclass** [phel-lang/phel-lang#156] [Ideas]
Without destructuring, we might do
```clojure
(def my-tuple @[:this :is :awesome])
(def lookup (get my-tuple 2))
(println lookup) # Prints :awesome
```

With destructuring; using let vs def
```clojure
(let [[_ _ c] my-tuple] (println c)) # Prints :awesome
(def [_ _ c] my-tuple) # Error: -> 'First argument of 'def must be a Symbol.'
(def [a b c] my-tuple) # Error: -> 'First argument of 'def must be a Symbol.'
```

What about to be able to deconstruct a tuple or a table using def? For example: 
```clojure
(def [_ _ c] @[1 2 3])
(println c) # Print 3

(def @{:b b} @{:a 1 :b 2 :c 3})
(println b) # Prints 2
```

I saw this functionality in [Janet](https://janet-lang.org/docs/destructuring.html), but I didn't see it in [Clojure](https://clojure.org/guides/destructuring)...

Does it make sense to implement this? What do you think, @jenshaase?

**Comment by jenshaase** [comment]
Well, that can be done but I think it bring not much benefits.

`def` is used to define a global variable. In practice you will not use them very often. It is more like a `const` in PHP. However, if you use the REPL it can make sense to do it.

Another reason is that you can add meta data to a `def`. With destructuring it would be unclear where the meta information belongs to.

Regarding Janet: The `def` in Janet is a bit different. `def` can be used to define global and local variables.

**Reply by Chemaclass** [reply]
I see. Thanks for the explanation. It's a non-prior for me, so we can leave this idea out (for now at least).
